File: parallelization/scripts/aggregate_results.R
cat("Aggregating results...\n")
RESULTS_PATH <- here::here("results")
method_names <- c("rf", "knn")

# load in results
preds <- purrr::map(
    method_names,
    ~ unlist(readRDS(here::here(RESULTS_PATH, sprintf("preds_%s.rds", .x))))
) |>
  setNames(method_names) |>
  dplyr::bind_cols()

# save aggregated results
write.csv(
  preds, 
  file = here::here(RESULTS_PATH, "preds_r.csv"),
  row.names = FALSE
)
cat("Done!\n")

File: parallelization/scripts/aggregate_results.py
from os.path import join as oj
import pandas as pd
import pickle as pkl


if __name__ == "__main__":
    print("Aggregating results...")
    RESULTS_PATH = oj("results")
    method_names = ["rf", "knn"]

    # Load the results
    results_dict = {}
    for method_name in method_names:
        results_dict[method_name] = pkl.load(
            open(oj(RESULTS_PATH, f"preds_{method_name}.pkl"), "rb")
        )
    results = pd.DataFrame(results_dict)

    # Save the aggregated results
    results.to_csv(oj(RESULTS_PATH, "preds_python.csv"), index=False)
    print("Done!")
    